### ISSUE TITLE
Brotli compilation failure

### ISSUE THREAD

**Issue by 1d2d3d** [microsoft/vcpkg#14574]
Hi! I'm getting a compile failure trying to install brotli using the latest vcpkg on Windows version 10.0.19041.
In an emscripten-enabled command shell, I issue:
    `vcpkg install brotli:wasm32-emscripten`
and get:
```Computing installation plan...
The following packages will be built and installed:
    brotli[core]:wasm32-emscripten
Detecting compiler hash for triplet wasm32-emscripten...
Starting package 1/1: brotli:wasm32-emscripten
Building package brotli[core]:wasm32-emscripten...
Could not locate cached archive: AppData\Local\vcpkg\archives\35\352df2258263de4bf89c44a3a2487cad06aa7e39.zip
-- Using community triplet wasm32-emscripten. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: vcpkg32\triplets\community\wasm32-emscripten.cmake
-- Using cached vcpkg32/downloads/google-brotli-e61745a6b7add50d380cfd7d3883dd6c62fc2c71.tar.gz
-- Cleaning sources at vcpkg32/buildtrees/brotli/src/6c62fc2c71-a8c4ea9278.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source vcpkg32/downloads/google-brotli-e61745a6b7add50d380cfd7d3883dd6c62fc2c71.tar.gz
-- Applying patch install.patch
-- Applying patch fix-arm-uwp.patch
-- Applying patch pkgconfig.patch
-- Using source at vcpkg32/buildtrees/brotli/src/6c62fc2c71-a8c4ea9278.clean
-- Configuring wasm32-emscripten
-- Building wasm32-emscripten-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:142 (message):
    Command failed: vcpkg32/downloads/tools/cmake-3.18.4-windows/cmake-3.18.4-win32-x86/bin/cmake.exe --build . --config Debug --target install -- -v -j5
    Working Directory: vcpkg32/buildtrees/brotli/wasm32-emscripten-dbg
    See logs for more information:
      vcpkg32\buildtrees\brotli\install-wasm32-emscripten-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:93 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/brotli/portfile.cmake:19 (vcpkg_install_cmake)
  scripts/ports.cmake:136 (include)


**Error: Building package brotli:wasm32-emscripten failed with: BUILD_FAILED**
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: brotli:wasm32-emscripten
  Vcpkg version: 2020.11.12-nohash

Additionally, attach any relevant sections from the log files above.
```
The install-wasm32-emscripten-dbg-err.log file contains:
   `ninja: error: unknown target 'install'`

**Comment by lwyj123** [comment]
Same issues here

**Comment by pgoodman** [comment]
I am trying to use brotli, not using wasm/emscripten, but instead use asan. What I've discovered is that the `check_c_source_compiles` check in brotli's CMakeLists.txt for detecting emscripten support seems to always succeed. I tried manually patching in `#error 1` and also a string of random symbols into the call to `check_c_source_compiles` and neither of these caused the `check_c_source_compiles` to fail.
